drivers/spi/spi_nxp_s32.c

```diff
@@ -0,0 +1,716 @@
+		irq_enable(DT_IRQN(SPI_NXP_S32_NODE(n)));				\
+	}
+
+#define SPI_NXP_S32_CONFIG_CALLBACK_FUNC(n)						\
+	.cb = spi_nxp_s32_##n##_callback,
+
+#define SPI_NXP_S32_CALLBACK_DEFINE(n)							\
+	static void spi_nxp_s32_##n##_callback(uint8 instance, Spi_Ip_EventType event)	\
+	{										\
+		ARG_UNUSED(instance);							\
+		const struct device *dev = DEVICE_DT_GET(SPI_NXP_S32_NODE(n));		\
+											\
+		spi_nxp_s32_transfer_callback(dev, event);				\
+	}
+#else
+#define SPI_NXP_S32_CONFIG_INTERRUPT_FUNC(n)
+#define SPI_NXP_S32_INTERRUPT_DEFINE(n)
+#define SPI_NXP_S32_CONFIG_CALLBACK_FUNC(n)
+#define SPI_NXP_S32_CALLBACK_DEFINE(n)
+#endif /*CONFIG_NXP_S32_SPI_INTERRUPT*/
+
+/*
+ * Declare the default configuration for SPI driver, no DMA
+ * support, all inner module Chip Selects are active low.
+ */
+#define SPI_NXP_S32_INSTANCE_CONFIG(n)							\
+	static const Spi_Ip_ConfigType spi_nxp_s32_default_config_##n = {		\
+		.Instance = n,								\
+		.Mcr = (SPI_MCR_MSTR(SPI_NXP_S32_IS_MASTER(n)) |			\
+			SPI_MCR_CONT_SCKE(0U) |	SPI_MCR_FRZ(0U) |			\
+			SPI_MCR_MTFE(0U) | SPI_MCR_SMPL_PT(0U) |			\
+			SPI_MCR_PCSIS(BIT_MASK(SPI_NXP_S32_NUM_CS(n))) |		\
+			SPI_MCR_MDIS(0U) | SPI_MCR_XSPI(1U) | SPI_MCR_HALT(1U)),	\
+		.TransferMode = SPI_IP_POLLING,						\
+		.StateIndex   = n,							\
+		SPI_NXP_S32_SET_SLAVE(n)						\
+	}
+
+#define SPI_NXP_S32_TRANSFER_CONFIG(n)							\
+	.transfer_cfg = {								\
+		.Instance = n,								\
+		.Ctare = SPI_CTARE_FMSZE(0U) | SPI_CTARE_DTCP(1U),			\
+	}
+
+#define SPI_NXP_S32_DEVICE(n)								\
+	PINCTRL_DT_DEFINE(SPI_NXP_S32_NODE(n));						\
+	SPI_NXP_S32_CALLBACK_DEFINE(n)							\
+	SPI_NXP_S32_INTERRUPT_DEFINE(n)							\
+	SPI_NXP_S32_INSTANCE_CONFIG(n);							\
+	static const struct spi_nxp_s32_config spi_nxp_s32_config_##n = {		\
+		.instance     = n,							\
+		.num_cs	      = SPI_NXP_S32_NUM_CS(n),					\
+		.clock_frequency = DT_PROP(SPI_NXP_S32_NODE(n), clock_frequency),	\
+		.sck_cs_delay = DT_PROP_OR(SPI_NXP_S32_NODE(n), spi_sck_cs_delay, 0U),	\
+		.cs_sck_delay = DT_PROP_OR(SPI_NXP_S32_NODE(n), spi_cs_sck_delay, 0U),	\
+		.cs_cs_delay  = DT_PROP_OR(SPI_NXP_S32_NODE(n), spi_cs_cs_delay, 0U),	\
+		.spi_hw_cfg   = (Spi_Ip_ConfigType *)&spi_nxp_s32_default_config_##n,	\
+		.pincfg = PINCTRL_DT_DEV_CONFIG_GET(SPI_NXP_S32_NODE(n)),		\
+		SPI_NXP_S32_CONFIG_CALLBACK_FUNC(n)					\
+		SPI_NXP_S32_CONFIG_INTERRUPT_FUNC(n)					\
+	};										\
+	static struct spi_nxp_s32_data spi_nxp_s32_data_##n = {				\
+		SPI_NXP_S32_TRANSFER_CONFIG(n),						\
+		SPI_CONTEXT_INIT_LOCK(spi_nxp_s32_data_##n, ctx),			\
+		SPI_CONTEXT_INIT_SYNC(spi_nxp_s32_data_##n, ctx),			\
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(SPI_NXP_S32_NODE(n), ctx)		\
+	};										\
+	DEVICE_DT_DEFINE(SPI_NXP_S32_NODE(n),						\
+			&spi_nxp_s32_init, NULL,					\
+			&spi_nxp_s32_data_##n, &spi_nxp_s32_config_##n,			\
+			POST_KERNEL, CONFIG_SPI_INIT_PRIORITY,				\
+			&spi_nxp_s32_driver_api);
+
+#if DT_NODE_HAS_STATUS(SPI_NXP_S32_NODE(0), okay)
```

Comment:
Please use `DT_INST_FOREACH_STATUS_OKAY`. You refer to other SPI drivers.

Reply:
The HAL is index based so all NXP S32 shim driver are using this approach. Already discussed in https://github.com/zephyrproject-rtos/zephyr/pull/49161#discussion_r949770066

Reply:
Thank you.